automation/orchestrated_pr_runner.py

```
@@ -0,0 +1,388 @@
+def sanitize_workspace_name(name: str, pr_number: int) -> str:
+    sanitized = re.sub(r"[^A-Za-z0-9._-]+", "-", name).strip("-")
+    return f"pr-{pr_number}-{sanitized}" if sanitized else f"pr-{pr_number}"
+
+
+def prepare_workspace_dir(repo: str, workspace_name: str) -> Path:
+    target = WORKSPACE_ROOT_BASE / repo / workspace_name
+    # Safety: ensure target stays within the configured root
+    try:
+        target.resolve().relative_to(WORKSPACE_ROOT_BASE.resolve())
+    except ValueError:
+        raise ValueError(f"Workspace path escapes root: {target}")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)